lib/snowflex/transport/http.ex

```diff
@@ -292,8 +292,24 @@ defmodule Snowflex.Transport.Http do
       {:ok, %{status: status, body: body}} when status in 200..299 ->
         {:ok, body}
 
-      {:ok, %{body: %{"code" => code, "message" => message}}} ->
-        {:error, %Error{message: String.replace(message, ~r/\n/, " "), code: code}}
+      {:ok, %{body: body}} ->
+        {:error,
+         %Error{
+           message: String.replace(body["message"] || "Unknown error", ~r/\n/, " "),
+           code: body["code"],
+           sql_state: body["sqlState"],
+           metadata:
+             %{
+               response: body,
+               query_id: statement_handle,
+               account_name: state.account_name,
+               username: state.username
+             }
+             |> maybe_put(:database, state.database)
+             |> maybe_put(:schema, state.schema)
+             |> maybe_put(:warehouse, state.warehouse)
+             |> maybe_put(:role, state.role)
```

Comment:
A couple of thoughts:
- I think that it is fine to have nil values in the metadata, no?
- If we really don't want to include it, I think that the lines in `Connection` do the work for removing `nil` entries.